lib/active_model/serializer/adapter/json.rb

```diff
@@ -6,45 +6,128 @@ class Adapter
       class Json < Adapter
         def serializable_hash(options = nil)
           options ||= {}
+          serialized_hash = {}
+
           if serializer.respond_to?(:each)
-            @result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            serialized_hash = serialize_array_without_root(serializer, options)
           else
-            @hash = {}
+            serialized_hash = resource_object_for(serializer)
+            serialized_hash = add_resource_relationships(serialized_hash, serializer)
+          end
+
+          # TODO: remove usage of result instance variable from other flows / tests
+          @result = serialized_hash
+
+          { root => serialized_hash }
+        end
+
+        def fragment_cache(cached_hash, non_cached_hash)
+          Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
+        end
 
-            @core = cache_check(serializer) do
-              serializer.attributes(options)
+        private
+
+        # iterate through the associations on the serializer,
+        # adding them to the parent as needed (as singular or plural)
+        #
+        # nested_associations is a list of symbols that governs what
+        # associations on the passed in seralizer to include
+        def add_resource_relationships(parent, serializer, nested_associations = {})
+          # have the include array normalized
+          nested_associations = ActiveModel::Serializer::Utils.include_options_to_hash(nested_associations)
```

Comment:
I tried that, but the options are on the association, rather than the render options.  would this be an appropriate test for that?

    render json: @something, include: {....stuff....}


I currently don't have any controller tests